common-tools/clas-detector/src/main/java/org/jlab/detector/pulse/ModeAHDC.java

```diff
@@ -0,0 +1,243 @@
+package org.jlab.detector.pulse;
+
+import java.util.List;
+import java.util.ArrayList;
+import org.jlab.utils.groups.NamedEntry;
+
+
+/**
+ * A new extraction method dedicated to the AHDC signal waveform
+ * 
+ * Some blocks of code are inspired by MVTFitter.java
+ *
+ * @author  ftouchte
+ */
+public class ModeAHDC extends HipoExtractor  {
+
+	// Settings parameters (they can be initialised by a CCDB)
+	public float samplingTime;
+	public int sparseSample;
+	public short adcOffset;
+	public long timeStamp;
+	public float fineTimeStampResolution;
+	public static final short ADC_LIMIT = 4095; // 2^12-1
+	public float amplitudeFractionCFA;
```

Comment:
All the pulse specific variables should be local variables in the extract method

Reply:
yes, good catch, critical for thread safety